### PR TITLE
Feature/1173 (3-phase Houses)

### PR DESCRIPTION
**Pull request by temcdrm**
We sometimes use houses to represent small commercial buildings, like strip malls, where each commercial HVAC zone is represented by a house.  Many small and medium commercial loads are actually connected to three-phase transformers, not single-phase transformers with center-tapped secondaries.  The three-phase transformer connection:

- Better represents the assets used by a utility
- Provides a one-for-one correspondence in connection points between EnergyPlus/Modelica building models and the house-zone models
- In some cases, a three-phase transformer will reduce the voltage drop and angle difference to more realistic values
- A three-phase transformer tends to balance the load current and reduce losses

This option allows the house to have a "load" as parent, i.e., not a triplex connection point.  In such cases, the nominal voltage defines an internal turns ratio for the house, such that panel loads are calculated as before at 120/240 volts. In other words, a 480-volt connection still looks like 120/240 to the house panel. When the panel is synched, the internally balanced ZIP equivalent loads are equally distributed among the phases present. Some limitations or untested conditions:

- The building load values are pushed line-to-neutral, with no option for line-to-line
- Two-phase connections have not been tested. There is a possible use case for these with open wye or open delta transformer connections, but these transformers are not in GridLAB-D yet, and they would appear as three-phase service to the building.
- Connections other than 120 (single-phase), 208 and 480 have not been tested.

The new test_commercial_meters.glm autotest illustrates the expected use cases for TESP and the DSOT study.

### CODE REVIEW

File: residential/house_e.cpp [hunk 1]
```diff
@@ -1349,6 +1393,56 @@ int house_e::init(OBJECT *parent)
 
 		//Set flag
 		proper_meter_parent = true;
+		commercial_load_parent = false;
+	}
+	else if (parent!=NULL && (gl_object_isa(parent,"load","powerflow"))) // for three-phase commercial zone-houses
```
Comment (afisher1) [inline residential/house_e.cpp]: For commercial zone-houses the parent should be a meter or a node. Not a load

**Comment by temcdrm** [comment]
It can work with "load" or "meter" as parent, but "node" doesn't have the ZIP attributes like "triplex_node" does.

**Comment by temcdrm** [comment]
There is one significant conflicting change in **configure.ac**, in that Boost was removed from the build for TESP with Helics.  Maybe this should be optional for GridLAB-D, because it can be a hassle to provide or build Boost, especially on non-Linux platforms. There is also a small, insignificant edit on a comment line in configure.ac.

Apart from that, feature/1173 seems to have settled and this would be a good time for TESP and DSOT to re-synch with the develop branch.

**Comment by eberleim** [comment]
@temcdrm Current versions of HELICS no longer use Boost, so unless TESP still uses Boost, it can be removed from both locations, and resolved. 
@ftuffner librt provides real-time support, for systems required to run in realtime environments. I'm unsure if GridLAB actually uses it in any way, or if it was linked in for some other reason. 

Let me know if you need any help reconciling the develop branch changes in so this can be done conflict-free.

**Comment by temcdrm** [comment]
We only had Boost in this branch for linking with HELICS, so it will be great to remove Boost...

File: configure.ac [hunk 2]
```diff
@@ -496,25 +506,25 @@ gld_save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $HELICS_CPPFLAGS"
...
+					[*lhelics*], [], [HELICS_LIBS="$HELICS_LIBS -lhelics-shared -lrt -lzmq"])
 				AS_CASE([$LIBS],
-					[*lhelics*], [], [LIBS="$LIBS -lhelics-shared -lrt -lzmq -lboost_filesystem -lboost_program_options -lboost_system -lboost_unit_test_framework"])],
+					[*lhelics*], [], [LIBS="$LIBS -lhelics-shared -lrt -lzmq"])],
 	[*darwin*], [AS_CASE([$HELICS_LIBS],
-					[*lhelics*], [], [HELICS_LIBS="$HELICS_LIBS -lhelics-shared -lrt -lzmq -lboost_filesystem -lboost_program_options -lboost_system -lboost_unit_test_framework"])
+					[*lhelics*], [], [HELICS_LIBS="$HELICS_LIBS -lhelicsSharedLib -lzmq"])
 				AS_CASE([$LIBS],
-					[*lhelics*], [], [LIBS="$LIBS -lhelics-shared -lrt -lzmq -lboost_filesystem -lboost_program_options -lboost_system -lboost_unit_test_framework"])],
+					[*lhelics*], [], [LIBS="$LIBS -lhelicsSharedLib -lzmq"])],
```
Comment (eberleim) [inline configure.ac]: `helicsSharedLib` is the C-linkages for HELICS, and should generally be avoided in C++ applications. 
the correct linkages to use now are `helics-shared` for release builds of HELICS, and `helics-sharedd` for debug builds. 
I believe all interfaces provided by `helicsSharedLib` should also be available through `helics-shared`, but will likely perform better when linked.

Comment (afisher1) [reply configure.ac]: I changed it to use helics-shared.

**Pull request merged**